nixos/modules/services/networking/freeradius.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.freeradius;
+
+    freeradiusService = cfg: optionalAttrs cfg.enable {
```

Comment:
The `optionalAttrs` is not needed because you already guard the config below with `mkIf`. Also, returning the systemd service attrset is nicer than returning `{ freeradius = { ... }; }` because the resulting service name can be chosen more elegantly.

Reply:
Also, the indenten is slightly off, Should be on the same level as `cfg`.

Reply:
Done.

I moved the `freeradius` name to the config section (not sure if that's where it should go). And I tested it in my setup.